### DISCUSSION TITLE
mwptools on MacOS

### DISCUSSION THREAD

**Discussion by stronnag** [stronnag/mwptools#188] [General]
# Installing mwptools on MacOS

## NOTICE

This post will NOT be updated. 
The definitive instructions are now in the [User Guide](https://stronnag.github.io/mwptools/mwp-macos/)

## OS requirements

A version of MacOS supporting `homebrew`, e.g.

### Apple Silicon

* Sequoia
* Sonoma
* Ventura

# X86_64

* Sonoma
* Ventura

Although Homebrew does not officially support  Sequoia for (most of) the packages required, the required packages (and mwp) work well on Sequoia.

## Install HomeBrew

```
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```

or use the official package from the Homebrew release area.

### Dependencies

```
brew install gtk4 \
 libshumate \
 libsoup \
 git \
 meson \
 libadwaita \
 libsecret \
 vala \
 protobuf-c \
 libpaho-mqtt \
 gstreamer \
 go \
 espeak-ng \
 vte3 \
 adwaita-icon-theme \
 gnuplot
```

### Blueprint compiler

This is not in `homebrew`, install locally:

```
git clone https://gitlab.gnome.org/jwestman/blueprint-compiler
cd blueprint-compiler/
meson setup _build
sudo ninja -C _build install
```

### Build mwp

Only local install is "supported"

* Clone the repository
* Checkout the **development** branch (for now)
* `cd mwptools`

```
meson setup _build --buildtype=release --strip --prefix=~/.local
ninja -C _build install
```

#### One off post install

* It is necessary to add `$HOME/.local/bin` to `$PATH`

mwp's `gsettings` does not work on MacOS (or more strictly, it does, but `mwp` and `gsettings` do not share storage, which is not helpful). Therefore mwp is using a `ini` file backend on MacOS.

* Prior to 24.12.28, it was necessary copy the template `ini` file into the config directory
  ```
  mkdir -p ~/.config/mwp/
  cp docs/mwp.ini ~/.config/mwp/
  ```
* To set settings outside of mwp, you may edit  `~/.config/mwp/mwp.ini` **with care** as there is no error checking.

## Other notes

macOS appears not measure text width in the same was a other OS. A "fudge factor" is included that works on my "Ventura" VM. The user set there own value with the environment variable `MWP_MAC_FACTOR', set to a value greater than 100. Try with values around 125.

e.g.
```
mwp MWP_MAC_FACTOR=128
```
Once a suitable value is found:

* Reset panel sizing `rm -f ~/.config/mwp/.paned`
* Add the environment variable to `~/.config/mwp/cmdopts`

## NOTICE

This post will NOT be updated. 
The definitive instructions are now in the [User Guide](https://stronnag.github.io/mwptools/mwp-macos/)

**Comment by nsstein** [comment]
Intel Mac running macos sequoia 15.2
 
 Along with the HomeBrew install dependencies, and Blueprint compiler
 I had to install  pip install PyGObject  (I am using pyenv)
 
 Other instructions were fine....
 
 I will start some testing as time frees up.  
 
 Thanks for the great headstart.  
 
 (and yes, it is time for a new computer)

**Reply by stronnag** [reply]
Interesting, that was not necessary on Ventura or Sonoma on Intel (VM). I will add it to the notes.
I note that a number of the dependencies are not officially supported by Homebrew on Sequoia.

**Reply by stronnag** [reply]
I will be interested to read your test experiences.

**Reply by nsstein** [reply]
it is probably due to my python setup which is related to a few other projects.    I plan in digging in more this weekend.

**Reply by stronnag** [reply]
Indeed, just done a clean install in a Sequoia VM, No problems with building / running `blueprint-compiler`. Note added to the User Guide about extant python installations.

**Comment by stronnag** [comment]
## Todo / Known Issues list

- [x] ~Audio (TTS) libraries are not recognised, however `--voice-command=espeak` does work~. Fixed in 488ca9b5.
- [ ] Media doesn't play, due to missing gtk `libmedia-gstreamer` shared library. No workaround at the moment.
- [x] Document / provide template for `mwp.app` (so it appears in `Applications`). See [User Guide](https://stronnag.github.io/mwptools/mwp-macos/) . Requires development branch 24.12.27 or later for files.
